extensions/rvadapter/src/main/java/com/google/android/agera/rvadapter/RepositoryPresenterCompilerStates.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.agera.rvadapter;
+
+import com.google.android.agera.Binder;
+import com.google.android.agera.Function;
+import com.google.android.agera.Receiver;
+import com.google.android.agera.Repository;
+import com.google.android.agera.Result;
+
+import android.support.annotation.LayoutRes;
+import android.support.annotation.NonNull;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+
+import java.util.List;
+
+/**
+ * Container of the compiler state interfaces supporting the creation of a
+ * {@link RepositoryPresenter}.
+ */
+public interface RepositoryPresenterCompilerStates {
```

Comment:
good point, I'll have to do some thinking on how to add this in a good way (maybe just another optional step in the compiler, but I kind of hope we can be smarter than that).
Something along the lines of a WeakHashMap with the data as (weak) keys and a simple counter to count up and add integers as values in place itemId? Would that work you think?